src/lib/db/segment-store.ts

```diff
@@ -97,8 +99,29 @@ export default class SegmentStore implements ISegmentStore {
 
     async getAll(): Promise<ISegment[]> {
         const rows: ISegmentRow[] = await this.db
-            .select(this.prefixColumns())
+            .select(
+                this.prefixColumns(),
+                'used_in_projects',
+                'used_in_features',
+            )
+            .countDistinct(
```

Comment:
have you tested it with a big data set in terms of performance?

Reply:
Analyzed this query on enterprise, it is just regular join, quite cheap.